src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/PackageUpdater.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater
+{
+    public class PackageUpdater
+    {
+        private readonly XDocument _doc;
+        private readonly ILogger _logger;
+
+        public PackageUpdater(XDocument doc, ILogger logger)
+        {
+            _doc = doc;
+            _logger = logger;
+        }
+
+        public XDocument UpdatePackages()
+        {
+            // removes references to System.ServiceModel
+            _doc.Root.Descendants("PackageReference")
+                .Where(x => x.Attribute("Include").Value.StartsWith("System.ServiceModel", System.StringComparison.OrdinalIgnoreCase))
+                .Remove();
+            _logger.LogDebug("Finish removing references to System.ServiceModel.");
+
+            // adds CoreWCF packages
+            var reference_group = _doc.Root.Descendants("PackageReference").Ancestors().First();
+            var corewcfPackages = XDocument.Parse(Constants.CoreWCFPackages);
```

Comment:
The IProjectFile would be really convenient for managing packages! But it would complicate the requirements to construct a  `PackageUpdater`. Now the constructor only requires a XDocument and it's easy to test. I am using the `GetFile()` method in `WCFUpdater.cs` to get easy access to project file and for the updaters it might be better to have some separation.